File: eLetter25.API/Auth/Models/AuthModels.cs
namespace eLetter25.API.Auth.Models;

/// <summary>
/// Request model for user registration
/// </summary>
/// <param name="Email">User's email address</param>
/// <param name="Password">Password</param>
/// <param name="EnableNotifications">Enable notifications</param>
public sealed record RegisterRequest(
    string Email,
    string Password,
    bool EnableNotifications = false);

/// <summary>
/// Request model for login
/// </summary>
/// <param name="Email">Email address</param>
/// <param name="Password">Password</param>
public sealed record LoginRequest(string Email, string Password);

/// <summary>
/// Response model for login
/// </summary>
/// <param name="AccessToken">JWT Access Token</param>
public sealed record LoginResponse(string AccessToken);

File: eLetter25.Client/eLetter25.Client/src/app/auth/models/auth.models.ts
export type { ApiError, ApiErrorResponse } from '../../core/models/api-error.models';

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  accessToken: string;
}

export interface RegisterRequest {
  email: string;
  password: string;
  enableNotifications: boolean;
}

export interface RegisterResponse {
  userId: string;
  message: string;
}
